core/src/test/java/org/testcontainers/utility/MountableFileTest.java

```diff
@@ -122,6 +122,26 @@ public void noTrailingSlashesInTarEntryNames() throws Exception {
         }
     }
 
+    @Test
+    public void equalsAndHashCode() throws Exception {
```

Comment:
could use more (parameterized) tests

Reply:
You mean with different file names and modes? Might be a bit over-engineered, considering the actual code is generated ;-)

Reply:
with/without resolved lazy getters, with different modes, with different paths

Reply:
I just could check the lazyness by reflection... hmmm....

Reply:
there is no need to check the lazyness. Test that the object is still equals after you call the getters

Reply:
Just added tests

Reply:
and another one: `equalsAndHashCodeDifferWithPath`